scripts/bootstrap/buildbootstrapcli.sh

```diff
@@ -0,0 +1,271 @@
+    echo "**** CLONING CORECLR REPOSITORY ****"
+    git clone https://github.com/dotnet/coreclr.git
+    cd coreclr
+    git checkout $__coreclrhash
+    cd ..
+fi
+
+if [[ ! -d corefx ]]; then
+    echo "**** CLONING COREFX REPOSITORY ****"
+    git clone https://github.com/dotnet/corefx.git
+    cd  corefx
+    git checkout $__corefxhash
+    cd ..
+fi
+
+if [[ ! -d core-setup ]]; then
+    echo "**** CLONING CORE-SETUP REPOSITORY ****"
+    git clone https://github.com/dotnet/core-setup.git
+    cd  core-setup
+    git checkout $__coresetuphash
+    cd ..
+fi
+
+echo "**** BUILDING CORE-SETUP NATIVE COMPONENTS ****"
+cd core-setup
+src/corehost/build.sh --arch "$__build_arch" --hostver "2.0.0" --apphostver "2.0.0" --fxrver "2.0.0" --policyver "2.0.0" --commithash `git rev-parse HEAD`
+cd ..
+
+echo "**** BUILDING CORECLR NATIVE COMPONENTS ****"
+cd coreclr
+./build.sh $__configuration $__build_arch $__clangversion 2>&1 | tee coreclr.log
+export __coreclrbin=$(cat coreclr.log | sed -n -e 's/^.*Product binaries are available at //p')
+cd ..
+echo "CoreCLR binaries will be copied from $__coreclrbin"
+
+echo "**** BUILDING COREFX NATIVE COMPONENTS ****"
+corefx/src/Native/build-native.sh $__build_arch $__configuration $__clangversion $__build_os 2>&1 | tee corefx.log
+export __corefxbin=$(cat corefx.log | sed -n -e 's/^.*Build files have been written to: //p')
+echo "CoreFX binaries will be copied from $__corefxbin"
+
+echo "**** Copying new binaries to dotnetcli/ ****"
+
+# First copy the coreclr repo binaries
+cp $__coreclrbin/*so $__frameworkpath
+cp $__coreclrbin/corerun $__frameworkpath
+cp $__coreclrbin/crossgen $__frameworkpath
+
+# Mark the coreclr executables as allowed to create executable memory mappings
+disable_pax_mprotect $__frameworkpath/corerun
+disable_pax_mprotect $__frameworkpath/crossgen
+
+# Now copy the core-setup repo binaries
+cp core-setup/cli/exe/dotnet/dotnet dotnetcli
+cp core-setup/cli/exe/dotnet/dotnet $__frameworkpath/corehost
+
+cp core-setup/cli/dll/libhostpolicy.so $__frameworkpath
+cp core-setup/cli/dll/libhostpolicy.so dotnetcli/sdk/$__sdkversion
+
+cp core-setup/cli/fxr/libhostfxr.so $__frameworkpath
+cp core-setup/cli/fxr/libhostfxr.so dotnetcli/host/fxr/$__fxrversion
+cp core-setup/cli/fxr/libhostfxr.so dotnetcli/sdk/$__sdkversion
+
+# Mark the core-setup executables as allowed to create executable memory mappings
+disable_pax_mprotect dotnetcli/dotnet
+disable_pax_mprotect $__frameworkpath/corehost
+
+# Finally copy the corefx repo binaries
+cp $__corefxbin/**/System.* $__frameworkpath
+
+# Copy System.Private.CoreLib.dll override from somewhere if requested
+if [[ "$__corelib" != "" ]]; then
+    cp "$__corelib" $__frameworkpath
+fi
+
+# Add the new RID to Microsoft.NETCore.App.deps.json
+# Replace the linux-x64 RID in the target, runtimeTarget and runtimes by the new RID
+# and add the new RID to the list of runtimes.
+echo "**** Adding new rid to Microsoft.NETCore.App.deps.json ****"
```

Comment:
At this point, I'd presume that you had enough bootstrapped to pass along the rid updating to an msbuild task if desired, correct?

Reply:
I am not sure I understand the question, could you please explain what you mean?

Reply:
You can ignore my comment.  sed seems like a reasonable implementation.

Slightly different question, at this point do you have enough of a bootstrapped cli that you can compile code?  Does Roslyn not need to be a part of the bootstrapping?  I could imagine that Roslyn has changed quite a bit since I last looked into bootstrapping, but previously I had to rebuild roslyn as well to really have something that could bootstrap.

Reply:
I've used this to bootstrap RHEL 6 and Alpine 3.6 from scratch. The cli that we get from this can be used to build managed parts of coreclr, corefx and core-setup and also to create / build / run apps. So after the script completes and there are no issues that need to be fixed, you can do e.g.
```
dotnet new console
dotnet run
```
And it builds and runs the hello world.
Of course there can be various potential issues based on the OS and architecture that is being brought up. These need to be discovered and fixed and it is hard to add a general guide.  
In general, if the above simple test works, then it is 99.999% really fully working.